### PR TITLE
feat: added music status component

### PR DESCRIPTION
**Pull request by AleGrz**
Added a fully configurable component that can display currently playing music and artist name. I closed #466 and squashed the branch, so now it should pass the checks. I'm still learning how to use git, sorry about that.

### CODE REVIEW

**Comment by Video-Nomad** [comment]
Tried this and it works great!
Some observations:

- While Spotify and other dedicated music apps update the play status instantly, chrome/edge/etc. does not. Not sure why, but it takes a while to switch from "paused" to "playing". Seems like a Windows thing, not the component itself.
- The label is not truncated. It would have been great to have a "max length" option to truncate it, similarly how window title component is truncated now (but not hardcoded).

File: README.md [hunk 1]
```diff
@@ -444,6 +444,17 @@ Use `Alt+Click` to pin and un-pin an icon.
   label_collapse_text: ">"
 ```
 
+### Bar Component: Music
+
+Displays currently playing music.
+
+```yaml
+- type: "musc"
```
Comment (Video-Nomad) [inline README.md:452]: Typo: "musc" -> "music"

File: README.md [hunk 2]
```diff
@@ -444,6 +444,17 @@ Use `Alt+Click` to pin and un-pin an icon.
...
+  label_not_playing: ""
+  label_paused: "{song_title} - {artist name}"
+  label_playing: "{song_title} - {artist name} ▶"
```
Comment (Lol-Lap) [inline README.md:455]:
```yaml
Typo: label_playing: "{song_title} - {artist name}" -> label_paused: "{song_title} - {artist_name}"
```

File: README.md [hunk 3]
```diff
@@ -444,6 +444,17 @@ Use `Alt+Click` to pin and un-pin an icon.
...
 ```
 
+### Bar Component: Music
+
+Displays currently playing music.
+
+```yaml
+- type: "musc"
+  label_not_playing: ""
+  label_paused: "{song_title} - {artist name}"
```
Comment (Lol-Lap) [inline README.md:454]: label_paused: "{song_title} - {artist name}" -> label_paused: "{song_title} - {artist_name}"

File: README.md (hunk 2 above)
Comment (Lol-Lap) [inline README.md:455]: label_playing: "{song_title} - {artist name}" -> label_paused: "{song_title} - {artist_name}"

Comment (Lol-Lap) [inline README.md:455]: label_playing: "{song_title} - {artist name}" -> label_playing: "{song_title} - {artist_name}"

**Comment by AleGrz** [comment]
I added trimming to title and artist individually, I think it looks better than truncating the whole label at once

**Comment by lars-berger** [comment]
Wow, this looks great @AleGrz 🙌 The code is excellent and it's working without issue on my end. Will go ahead and merge

**Pull request merged**